### PR TITLE
Adding a script to speed up cluster backups on enterprise

### PR DESCRIPTION
**Pull request by sridharavinash**
Only affects GHE cluster instances. No change for single appliance GHE.
This will not impact GHE instances older than 2.6.
You will need to upgrade to GHE 2.6.8 and 2.7.3 for this enhancement to be available.

This will benefit cluster setups with 3 or more nodes.

Some basic benchmarks:

This test is with roughly 1.2k repos of a total size of 14G.

Legacy script

```
real    4m2.751s
user    1m11.998s
sys     0m44.079s
Repo sync with no pre-calculated routes took 241s
```

New script backing up the repositories

```
real    2m3.509s
user    0m57.403s
sys     0m31.405s

* Transferring 297 repositories from ghe-test-dgit-fs1
* Transferring 304 repositories from ghe-test-dgit-fs2
* Transferring 268 repositories from ghe-test-dgit-fs3
* Transferring 280 repositories from ghe-test-dgit-fs4

Calculating sync routes took 4s
Repo sync with calculated routes took 118s
```

### CODE REVIEW

**Comment by sridharavinash** [comment]
@rubiojr back-ported some of the bug fixes from https://github.com/github/backup-utils-private/issues/61

Let me know if it looks good. 

Thanks

File: share/github-backup-utils/ghe-backup-repositories-cluster-ng [hunk 1]
```diff
@@ -0,0 +1,379 @@
...
+
+# rsync all the repositories
+bm_start "$(basename $0) - Repo sync"
+for file_list in $tempdir/*.rsync; do
+  hostname=$(basename $file_list .rsync)
+
+  repo_num=$(cat $file_list | wc -l)
+  echo "* Transferring $repo_num repositories from $hostname"
+
+  sync_data $hostname $file_list &
```
Comment (sbryant) [inline share/github-backup-utils/ghe-backup-repositories-cluster-ng]: this N seems like it should be small? Like only 3 or 4 processes given the file lists right?

File: share/github-backup-utils/ghe-backup-repositories-cluster-ng [hunk 2]
```diff
@@ -0,0 +1,379 @@
...
+done
+
+if ! test -f $tempdir/archived_repos.rsync; then
+  echo "* No archived repositories found to backup."
+  exit 0
+fi
+
+for h in $hostnames; do
+  echo "* Transferring archived routes from $h"
+  sync_data $h $tempdir/archived_repos.rsync 2>/dev/null || true
```
Comment (sbryant) [inline share/github-backup-utils/ghe-backup-repositories-cluster-ng]: can this be done in parallel?

Comment (sridharavinash) [inline share/github-backup-utils/ghe-backup-repositories-cluster-ng]: @sbryant Yes this is planned for a future update when we have some of @mutle archived repos ready. It'll use the same strategy that we use for the non-archived repos.

**Pull request merged**